packages/flutter/test/material/bottom_sheet_test.dart

```diff
@@ -681,6 +681,51 @@ void main() {
     );
   });
 
+  testWidgets('modal BottomSheet does not override MediaQuery when overrideMediaQuery is false', (WidgetTester tester) async {
+    late BuildContext outerContext;
+    late BuildContext innerContext;
+
+    await tester.pumpWidget(Localizations(
+      locale: const Locale('en', 'US'),
+      delegates: const <LocalizationsDelegate<dynamic>>[
+        DefaultWidgetsLocalizations.delegate,
+        DefaultMaterialLocalizations.delegate,
+      ],
+      child: Directionality(
+        textDirection: TextDirection.ltr,
+        child: MediaQuery(
+          data: const MediaQueryData(
+            padding: EdgeInsets.all(50.0),
+            size: Size(400.0, 600.0),
+          ),
+          child: Navigator(
+            onGenerateRoute: (_) {
+              return PageRouteBuilder<void>(
+                pageBuilder: (BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation) {
+                  outerContext = context;
+                  return Container();
+                },
+              );
+            },
+          ),
+        ),
+      ),
+    ));
+
+    showModalBottomSheet<void>(
+      context: outerContext,
+      overrideMediaQuery: false,
+      builder: (BuildContext context) {
+        innerContext = context;
+        return Container();
```

Comment:
Would this not also fix this? 

```suggestion
        return MediaQuery(
          data: MediaQuery.of(context)
          child: Container()
        );
```

I think adding a flag to override the inner workings of a widget is not a scalable pattern across the framework, and not very different from the solution above, especially in terms of being discoverable to the user. I haven't been able to find another case of `overrideMediaQuery`, but this was likened to the `Scaffold.extendBody` in the linked issue. Maybe just using the same naming convention will make it easier for the user to associate its purpose.

Reply:
> Would this not also fix this?

Did you meant `MediaQuery.of(outerContext)`? If yes, this is one of the workaround for this issue: storing the parent context, as done in this unit test. The problem is that this workaround is not easily discoverable and relies on a good understanding of BuildContext :wink: 

> I think adding a flag to override the inner workings of a widget is not a scalable pattern across the framework

I agree too. The problem here is that `BottomSheet` inner workings relies on something that few widget do : consuming padding using a call to `MediaQuery.removePadding`. Doing so, it breaks the inner `SafeArea` that a user might want to add (and it is very difficult for the user to find why composition with a `SafeArea` does not work inside a `BottomSheet`). I think the Flutter way would have been to let the user use composition to decide if he wants to consume the top padding. Just my 2 cents to explain why this fix is mainly a way to find an official workaround without breaking existing code. Otherwise, users will use workarounds such as this one: [adding an inner MaterialApp!](https://github.com/flutter/flutter/issues/48677#issuecomment-739907281). 

Adding a property is the good option for compatibility and I agree that the main point is to find a good name for it. I agree too that `overridesMediaQuery` is not very good and unusual. `extentBehingNavigationBar` will be on par with Scaffold similar property, but discoverability is not optimal and this name is not fully self explanatory in the BottomSheet case. Naming is hard :smile: 

I just remembered that `showDialog` has a `useSafeArea` property. We might do something similar here. Its default value would be false. If true, instead of adding a `MediaQuery` that consumes the topPadding, we can insert a `SafeArea`. It will also make `showModalBottomSheet` and `showDialog` similar. And I think it is great for discoverability because most of the users participating in those issues were looking for a way to add a `SafeArea` in their `BottomSheet`.

@Piinks I could try to implement this, do you think adding a `useSafeArea` property is a valuable approach?

Reply:
Well said! I do appreciate how thorough you always are. :)
I think `useSafeArea` makes sense so we have a consistent patterns for `showModalBottomSheet` and `showDialog`.

Can you try it with the test cases reported in the issue? I think this sounds like a great solution.

Reply:
@Piinks 
I updated the PR according to our previous discussion: adding a `useSafeArea` parameter to `showModalBottomSheet`.

About persistent bottom sheets, shown using `showBottomSheet` or `Scaffold.bottomSheet` property, despite both implementations adding a `BottomSheet` instance, the way they insert this widget in the tree is very different. The unusual `MediaQuery.removePadding` call is only done when calling `showModalBottomSheet` :

https://github.com/flutter/flutter/blob/5a5d021e51ca9a16fdfc64b34a8d64c75be1b500/packages/flutter/lib/src/material/bottom_sheet.dart#L524

There is one open issue, https://github.com/flutter/flutter/issues/69676, related to `showBottomSheet` and `SafeArea`. After investigating it, I think it is more actionable to not cover these two different cases in the same PR.

Reply:
Great! Thanks for investigating!